proposed/http-factory/http-factory.md

```diff
@@ -163,14 +170,17 @@ fwrite($resource, $body);
 Has the ability to create streams for uploaded files.
 
 ```php
-namespace Psr\Http\Message;
+namespace Psr\Http\Factory;
+
+use Psr\Http\Message\UploadedFileInterface;
 
 interface UploadedFileFactoryInterface
 {
     /**
      * Create a new uploaded file.
      *
-     * If a string is passed it is assumed to be a file path.
+     * If a string is used to create the file, a temporary resource will be
+     * created with the content of the string.
```

Comment:
I foresee a use case within async (non-SAPI) servers, where they may want to create an array of such instances to push into a `ServerRequestInterface`.

Another possibility is when parsing the request body to build the instances, as might happen in non-POST, non-form-encoded requests; having the factory allows middleware that performs this logic to do so without knowledge of the PSR-7 implementation.